cell_classification/augmentation_pipeline.py

```diff
@@ -137,3 +138,351 @@ def py_aug(batch, tf_aug):
+                labels = tf.expand_dims(labels, axis=0)
+                squeeze = True
+            _, h, w, _ = image.shape
+            image = transform(
+                image,
+                get_zoom_matrix(zooms, h, w),
+                fill_mode=self.fill_mode,
+                interpolation="bilinear",
+            )
+            labels = transform(
+                labels,
+                get_zoom_matrix(zooms, h, w),
+                fill_mode=self.fill_mode,
+                interpolation="nearest",
+            )
+            if squeeze:
+                image = tf.squeeze(image, axis=0)
+                labels = tf.squeeze(labels, axis=0)
+        return image, labels
+
+
+class LinearContrast(tf.Module):
+    """ Linear contrast augmentation"""
+    def __init__(self, prob=0.5, min_factor=0.5, max_factor=1.5):
+        """
+        Args:
+            prob (float):
+            The probability of applying the augmentation
+        min_factor (float):
+            The minimum contrast factor
+        max_factor (float):
+            The maximum contrast factor
+        """
+        super(LinearContrast, self).__init__()
+        self.prob = prob
+        self.min_factor = min_factor
+        self.max_factor = max_factor
+
+    def __call__(self, image, labels):
+        """ Apply linear contrast augmentation
+        Args:
+            image (tf.Tensor):
+                The image to augment
+                labels (tf.Tensor):
+                The labels to augment
+        Returns:
+            tf.Tensor:
+                The augmented image
+            tf.Tensor:
+                The augmented labels
+        """
+        if tf.random.uniform(()) < self.prob:
+            factor = tf.random.uniform((), self.min_factor, self.max_factor)
+            image *= factor
+        return image, labels
+
+
+class MixUp(tf.Module):
+    """ MixUp augmentation for segmentation data"""
+    def __init__(self, prob=0.5, alpha=0.2):
+        """
+        Args:
+            prob (float):
+                The probability of applying the augmentation
+            alpha (float):
+                The alpha parameter of the beta distribution
+        """
+        super(MixUp, self).__init__()
+        self.prob = prob
+        self.alpha = alpha
+
+    def __call__(self, image, labels):
+        """ Apply mixup augmentation
+        Args:
+            image (tf.Tensor):
+                The batch of images to augment
+            labels (tf.Tensor):
+                The batch of labels to augment
+        Returns:
+            tf.Tensor:
+                The augmented image batch
+            tf.Tensor:
+                The augmented label batch
+        """
+        labels = tf.cast(labels, dtype=tf.float32)
+        undecided = tf.where(labels == 2, True, False)
+        b = tf.shape(image)[0]
+        prob = tf.random.uniform([b], 0, 1) < self.prob
+        beta = tf.random.uniform([b], 0, 1)
```

Comment:
Some more comments here would be useful to explain how the vector `b` is being used to determine which images are augmented, the transformations to get the desired distribution, etc

Reply:
100%, the mixup function is not really readable atm.